consumers/writers/influxdb/consumer_test.go

```diff
@@ -12,64 +13,109 @@ import (
+	streamsSize   = 250
+	rowCountSenml = fmt.Sprintf(`from(bucket: "%s") 
+	|> range(start: -1h, stop: 1h) 
+	|> filter(fn: (r) => r["_measurement"] == "messages")
+	|> filter(fn: (r) => r["_field"] == "dataValue" or r["_field"] == "stringValue" or r["_field"] == "value" or r["_field"] == "boolValue" or r["_field"] == "sum" )
+	|> group(columns: ["_measurement"])
+	|> count()
+	|> yield(name: "count")`, repoCfg.Bucket)
+
+	rowCountJson = fmt.Sprintf(`from(bucket: "%s")
+	|> range(start: -1h, stop: 1h)
+	|> filter(fn: (r) => r["_measurement"] == "some_json")
+	|> filter(fn: (r) => r["_field"] == "field_1" or r["_field"] == "field_2" or r["_field"] == "field_3" or r["_field"] == "field_4" or r["_field"] == "field_5/field_1" or r["_field"] == "field_5/field_2")
+	|> count()
+	|> yield(name: "count")`, repoCfg.Bucket)
 	subtopic = "topic"
-)
 
-var (
+	client  influxdb2.Client
 	v       float64 = 5
 	stringV         = "value"
 	boolV           = true
 	dataV           = "base64"
 	sum     float64 = 42
+	repoCfg         = writer.RepoConfig{
+		Bucket: dbBucket,
+		Org:    dbOrg,
+	}
+	errUnexpectedType = errors.New("Unexpected response type")
 )
 
-// This is utility function to query the database.
-func queryDB(cmd string) ([][]interface{}, error) {
-	q := influxdata.Query{
-		Command:  cmd,
-		Database: testDB,
+func deleteBucket() error {
+	bucketsAPI := client.BucketsAPI()
+	bucket, err := bucketsAPI.FindBucketByName(context.Background(), repoCfg.Bucket)
+	if err != nil {
+		return err
 	}
-	response, err := client.Query(q)
+	err = bucketsAPI.DeleteBucket(context.Background(), bucket)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	if response.Error() != nil {
-		return nil, response.Error()
+	return nil
+}
+func createBucket() error {
+	orgAPI := client.OrganizationsAPI()
+	org, err := orgAPI.FindOrganizationByName(context.Background(), repoCfg.Org)
+	if err != nil {
+		return err
 	}
-	if len(response.Results[0].Series) == 0 {
-		return nil, nil
+	bucketsAPI := client.BucketsAPI()
+	_, err = bucketsAPI.CreateBucketWithName(context.Background(), org, repoCfg.Bucket)
+	if err != nil {
+		return err
 	}
-	// There is only one query, so only one result and
-	// all data are stored in the same series.
-	return response.Results[0].Series[0].Values, nil
+	return nil
+}
+func resetBucket() error {
+	if err := deleteBucket(); err != nil {
+		return err
+	}
+	if err := createBucket(); err != nil {
+		return err
+	}
+	return nil
 }
 
-func TestSaveSenml(t *testing.T) {
-	repo := writer.New(client, testDB)
+func queryDB(fluxQuery string) (int, error) {
+	rowCount := 0
+	queryAPI := client.QueryAPI(repoCfg.Org)
 
+	// get QueryTableResult
+	result, err := queryAPI.Query(context.Background(), fluxQuery)
+	if err != nil {
+		return rowCount, err
+	}
+	if result.Next() {
+		if value, ok := result.Record().Value().(int64); !ok {
+			return rowCount, errUnexpectedType
+		} else {
+			rowCount = int(value)
+		}
```

Comment:
No need for else here:
```
		if value, ok := result.Record().Value().(int64); !ok {
			return rowCount, errUnexpectedType
		}
	        rowCount = int(value)
```

Reply:
Here it says undeclared name: value
`rowCount = int(value)`


If I declare it above, It highlights this line and says value is declared but not used.
`if value, ok := result.Record().Value().(int64); !ok {`

Reply:
how yes sorry, in that case don't inline